cmake/Modules/UseODB.cmake

```diff
@@ -135,33 +141,55 @@ function(odb_compile outvar)
 	file(REMOVE_RECURSE "${ODB_COMPILE_OUTPUT_DIR}")
 	file(MAKE_DIRECTORY "${ODB_COMPILE_OUTPUT_DIR}")
 
-	foreach(input ${PARAM_FILES})
-		get_filename_component(fname "${input}" NAME_WE)
-		set(outputs)
-
-		foreach(sfx ${ODB_COMPILE_FILE_SUFFIX})
-			string(REGEX REPLACE ":.*$" "" pfx "${sfx}")
-			string(REGEX REPLACE "^.*:" "" sfx "${sfx}")
-
-			if(NOT "${PARAM_MULTI_DATABASE}" MATCHES "static" OR NOT "${pfx}" MATCHES "common")
-				set(output "${ODB_COMPILE_OUTPUT_DIR}/${fname}${sfx}${ODB_COMPILE_SOURCE_SUFFIX}")
-				list(APPEND ${outvar} "${output}")
-				list(APPEND outputs "${output}")
-			endif()
-		endforeach()
-
-		if(ODB_COMPILE_DEBUG)
-			set(_msg "${ODB_EXECUTABLE} ${ODB_ARGS} ${input}")
-			string(REPLACE ";" " " _msg "${_msg}")
-			message(STATUS "${_msg}")
-		endif()
-
-		add_custom_command(OUTPUT ${outputs}
-			COMMAND ${ODB_EXECUTABLE} ${ODB_ARGS} "${input}"
-			DEPENDS "${input}"
-			WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
-			VERBATIM)
-	endforeach()
+  if(PARAM_INPUT_NAME)
+    get_filename_component(fname "${PARAM_INPUT_NAME}" NAME_WE)
+            set(outputs)
+
+	  foreach(sfx ${ODB_COMPILE_FILE_SUFFIX})
+		  string(REGEX REPLACE ":.*$" "" pfx "${sfx}")
+		  string(REGEX REPLACE "^.*:" "" sfx "${sfx}")
+
+		  if(NOT "${PARAM_MULTI_DATABASE}" MATCHES "static" OR NOT "${pfx}" MATCHES "common")
+			  set(output "${ODB_COMPILE_OUTPUT_DIR}/${fname}${sfx}${ODB_COMPILE_SOURCE_SUFFIX}")
+			  list(APPEND ${outvar} "${output}")
+			  list(APPEND outputs "${output}")
+		  endif()
+	  endforeach()
+
+    add_custom_command(OUTPUT ${outputs}
+      COMMAND ${ODB_EXECUTABLE} ${ODB_ARGS} ${PARAM_FILES}
+      DEPENDS ${PARAM_FILES}
+            WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
+            VERBATIM)
+  else()
+	  foreach(input ${PARAM_FILES})
+		  get_filename_component(fname "${input}" NAME_WE)
+		  set(outputs)
+
+		  foreach(sfx ${ODB_COMPILE_FILE_SUFFIX})
+			  string(REGEX REPLACE ":.*$" "" pfx "${sfx}")
+			  string(REGEX REPLACE "^.*:" "" sfx "${sfx}")
+
+			  if(NOT "${PARAM_MULTI_DATABASE}" MATCHES "static" OR NOT "${pfx}" MATCHES "common")
+				  set(output "${ODB_COMPILE_OUTPUT_DIR}/${fname}${sfx}${ODB_COMPILE_SOURCE_SUFFIX}")
+				  list(APPEND ${outvar} "${output}")
+				  list(APPEND outputs "${output}")
+			  endif()
+		  endforeach()
+
+		  if(ODB_COMPILE_DEBUG)
+			  set(_msg "${ODB_EXECUTABLE} ${ODB_ARGS} ${input}")
+			  string(REPLACE ";" " " _msg "${_msg}")
+			  message(STATUS "${_msg}")
+		  endif()
+
+		  add_custom_command(OUTPUT ${outputs}
+			  COMMAND ${ODB_EXECUTABLE} ${ODB_ARGS} "${input}"
+			  DEPENDS "${input}"
+			  WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
+			  VERBATIM)
+	  endforeach()
+  endif()
 
```

Comment:
The intendation of this file was messed up, making it nearly impossible to compare.
Please use tabs for intendation.